lib/elixir/lib/file.ex

```diff
@@ -361,6 +361,39 @@ defmodule File do
   end
 
   @doc """
+  Reads the symbolic link at `path`. If path exists and is a symlink, returns `{:ok, target}`,
+  otherwise returns `{:error, reason}`.
+
+  For more details, see [`:file.read_link/1`](http://erlang.org/doc/man/file.html#read_link-1)
+
+  Typical error reasons are:
+    * `:einval` - path is not a symbolic link
+    * `:enoent` - path does not exist
+    * `:enotsup` - symbolic links are not supported on the current platform
+  """
+  @spec read_link(Path.t) :: {:ok, binary} | {:error, posix}
+  def read_link(path) do
+    case path |> IO.chardata_to_string |> F.read_link do
+      {:ok, target} -> {:ok, target |> IO.chardata_to_string}
+      error -> error
+    end
+  end
+
+  @doc """
+  Same as `read_link/1` but returns the target directly and throws `File.Error` if an error is
```

Comment:
@shribe "or" vs "and" was not about shortness, but about the semantics of it (that is, it either returns the target *or* throws, it doesn't do both 😛). :)